thorvg/examples/thorvg-example/main/display_main.c

```diff
@@ -0,0 +1,315 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include <stdlib.h>
+#include <dirent.h>
+#include "esp_log.h"
+#include "esp_err.h"
+#include "esp_check.h"
+#include "esp_timer.h"
+#include "esp_lcd_panel_ops.h"
+#include "bsp/esp-bsp.h"
+#include "driver/ppa.h"
+#include "thorvg_capi.h"
+
+static const char *TAG = "example";
+
+static void play_tick_new(esp_timer_handle_t *tm);
+static void play_tick_del(esp_timer_handle_t tm);
+static uint32_t play_tick_get(void);
+static uint32_t play_tick_elaps(uint32_t prev_tick);
+
+static void capi_loop_task(void *arg);
+static esp_err_t capi_create_lottie(ppa_client_handle_t ppa_handle, bsp_lcd_handles_t *lcd_panel);
+static esp_err_t argb888_to_rgb565_ppa(ppa_client_handle_t ppa_handle, const uint32_t *in, uint16_t *out);
+
+/* SPIFFS mount root */
+#define FS_MNT_PATH             BSP_SPIFFS_MOUNT_POINT
+
+#define LOTTIE_SIZE_HOR         (320)
+#define LOTTIE_SIZE_VER         (320)
+
+#define EXPECTED_FPS            (20)
+#define LOTTIE_FILENAME         FS_MNT_PATH"/emoji-animation.json"
+
+static uint32_t sys_time = 0;
+static volatile uint8_t tick_irq_flag;
+
+void app_main(void)
+{
+    static bsp_lcd_handles_t lcd_panel;
+
+    bsp_spiffs_mount();
+
+    /* Initialize display */
+    bsp_display_new_with_handles(NULL, &lcd_panel);
+
+    bsp_display_backlight_on();
+
+    BaseType_t res = xTaskCreate(capi_loop_task, "thorvg task", 60 * 1024, (void *)&lcd_panel, 5, NULL);
+    if (res != pdPASS) {
+        ESP_LOGE(TAG, "Create thorvg task fail!");
+    }
+}
+
+void capi_loop_task(void *arg)
+{
+    bsp_lcd_handles_t *lcd_panel = (bsp_lcd_handles_t *)arg;
+
+    ppa_client_handle_t ppa_client_srm_handle = NULL;
+    ppa_client_config_t ppa_client_config = {
+        .oper_type = PPA_OPERATION_SRM,
+    };
+    ppa_register_client(&ppa_client_config, &ppa_client_srm_handle);
+
+    if (ppa_client_srm_handle) {
+        while (1) {
+            capi_create_lottie(ppa_client_srm_handle, lcd_panel);
+            vTaskDelay(pdMS_TO_TICKS(2000));
+        }
+    } else {
+        ESP_LOGE(TAG, "ppa_register_client failed");
+    }
+
+    if (ppa_client_srm_handle) {
+        ppa_unregister_client(ppa_client_srm_handle);
+    }
+
+    vTaskDelete(NULL);
+}
+
+static esp_err_t capi_create_lottie(ppa_client_handle_t ppa_handle, bsp_lcd_handles_t *lcd_panel)
+{
+    esp_err_t ret = ESP_OK;
+
+    Tvg_Result tvg_res = TVG_RESULT_SUCCESS;
+    Tvg_Result tvg_engine = TVG_RESULT_INSUFFICIENT_CONDITION;
+
+    static uint32_t reac_color = 0x00;
+
+    uint32_t *canvas_buf_888 = NULL;
+    uint16_t *canvas_buf_565 = NULL;
+
+    Tvg_Animation *animation = NULL;
+    Tvg_Canvas *canvas = NULL;
+    esp_timer_handle_t play_timer = NULL;
+
+    play_tick_new(&play_timer);
+    ESP_GOTO_ON_FALSE(play_timer, ESP_ERR_INVALID_STATE, err, TAG, "Error create play_timer");
```

Comment:
because there're ESP_ERROR_CHECK in the `play_tick_new`, so when that function returns successfully, the `play_timer` must be non-null.